[release/5.0] Update nuget versions

wtgodbe opened the pull request:
Backport of https://github.com/dotnet/aspnetcore/pull/31064. Should fix .nuspec's so they all use `net5.0` instead of `.NETCoreApp5.0`

Related: discussion that starts here https://github.com/dotnet/aspnetcore/pull/29934#discussion_r605947867

wtgodbe committed:
Update nuget/msbuild versions

wtgodbe committed:
Update nuget

wtgodbe committed:
Nuget only

dougbu commented:
Please update BaselineGenerator.csproj here too. Up to you whether we redo baseline generation in this PR or just wait for the next round of branding.

wtgodbe commented:
@dougbu see https://github.com/dotnet/aspnetcore/pull/29934#issuecomment-812709598 - updating to 5.8.1 brings a whole other set of changes that are orthogonal to the behavior causing us issues. We could update 5.8.1, but we'd have to make another set of changes to the Baseline Generator to get that in.

wtgodbe committed:
Fix baseline, update to 5.8.1

wtgodbe commented:
Never mind - per the other PR, I updated to 5.8.1 and fixed the baseline: https://github.com/dotnet/aspnetcore/pull/31490/commits/a679991aa214e265e5ace69b8a74520e8f5e035a

wtgodbe commented:
all `.nuspec` files now look good (`net5.0` is the TFM where applicable)

wtgodbe committed:
Undo test change

dougbu reviewed (approved):
Any idea why the test changes are so minimal❔

wtgodbe committed:
Update MetadataSerializerTest.cs

wtgodbe commented:
> Any idea why the test changes are so minimal❔

Fixed, I had meant to undo all test changes
